runtime/server/queries_tables.go

```diff
@@ -39,15 +40,32 @@ type ColumnInfo struct {
 }
 
 var DoubleQuotesRegexp *regexp.Regexp = regexp.MustCompile("\"")
+var HyphenRegexp *regexp.Regexp = regexp.MustCompile("-")
 
 func EscapeDoubleQuotes(column string) string {
 	return DoubleQuotesRegexp.ReplaceAllString(column, "\"\"")
 }
 
+func EscapeHyphen(column string) string {
+	return HyphenRegexp.ReplaceAllString(column, "_")
+}
+
 func (s *Server) ProfileColumns(ctx context.Context, req *runtimev1.ProfileColumnsRequest) (*runtimev1.ProfileColumnsResponse, error) {
+	temporaryTableName := "profile_columns_" + EscapeHyphen(uuid.New().String())
+	// views return duplicate column names, so we need to create a temporary table
 	rows, err := s.query(ctx, req.InstanceId, &drivers.Statement{
+		Query:    fmt.Sprintf(`CREATE TEMPORARY TABLE "%s" AS (SELECT * FROM "%s" LIMIT 1)`, temporaryTableName, req.TableName),
+		Priority: int(req.Priority),
+	})
+	if err != nil {
+		return nil, err
+	}
+	rows.Close()
+	defer s.dropTempTable(ctx, req.InstanceId, int(req.Priority), temporaryTableName)
```

Comment:
Actually, all places that have a `dropTempTable` should use `context.Background()` instead of `ctx` to ensure cleanup even for cancelled requests.

Reply:
Good catch, thanks